### PR TITLE
Enabled appveyor for Windows build

### PR DESCRIPTION
**Pull request by hsbt**
(This is work in progress status yet)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+clone_depth: 10
+install:
+  - '"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" amd64'
+  - SET PATH=C:\Ruby%ruby_version%\bin;%PATH%
+  - gem install rake-compiler
```
Comment (rhenium) [inline appveyor.yml]: I think we should extract to a rake task (I'm currently using a dirty workaround in ruby-openssl-docker: https://github.com/ruby/openssl/blob/f542fb60add7a7f9f57bd7097a5e61dc3e5e0a99/tool/ruby-openssl-docker/init.sh#L18)

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+clone_depth: 10
+install:
+  - '"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" amd64'
```
Comment (hsbt) [inline appveyor.yml]: Your comment is correct. I will remove this.

File: appveyor.yml (hunk 1 above)
Comment (hsbt) [inline appveyor.yml]: I agree. I try to extract it after test success on windows build.

**Comment by hsbt** [comment]
fmm, I wonder that `openssl.so` couldn't reference openssl library on appveyor test phase. I investigate now.

**Comment by hsbt** [comment]
Everything works. I squashed WIP commits.

I make an another pull request for dependency installation like `rake-compiler` after this pull request merged.

**Comment by rhenium** [comment]
Thank you for the great work!

Regarding the dependency installation task, I tried here: https://gist.github.com/rhenium/4b08f402a00d316aa588f2a23f9c4916

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+clone_depth: 10
+install:
+  - SET PATH=C:\Ruby%ruby_version%\bin;%PATH%
+  - appveyor DownloadFile http://dl.bintray.com/oneclick/OpenKnapsack/x64/openssl-1.0.1l-x64-windows.tar.lzma
```
Comment (rhenium) [inline appveyor.yml]: Why 1.0.1l? It is fairly outdated... (Official support for 1.0.1 branch will cease at the end of this year.)

Comment (hsbt) [inline appveyor.yml]: Because Oneclick Installer uses 1.0.1 version. I tried to upgrade it to 1.1.0x.

Comment (rhenium) [inline appveyor.yml]: I see. It's unfortunate. Maybe we can update it when the next release of RubyInstaller.

Comment (hsbt) [inline appveyor.yml]: I agreed. I try to build openssl-1.1.0 binary used RubyInstaller recipe as https://github.com/oneclick/knapsack-recipes/tree/master/openssl later.

**Comment by hsbt** [comment]
I merge this. Continue to simplify preparation instructions.

**Pull request merged**